verification/verify.go

```diff
@@ -103,7 +112,29 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 		return nil, nil
 	}
 
-	// TODO sig checking; extract from broadcast.go
+	if params.Results != nil {
+		segHashes := make([][]byte, len(params.Results.Segments))
+		for i, segment := range params.Results.Segments {
+
+			data, err := drivers.GetSegmentData(segment.Url)
+			if err != nil {
+				glog.Errorf("%v error with segment: %v (URL: %v)", monitor.SegmentTranscodeErrorDownload, err, segment.Url)
+				break
+			}
+			segHashes[i] = crypto.Keccak256(data)
+		}
+
+		// Might not have seg hashes if results are directly uploaded to the broadcaster's OS
+		// TODO: Consider downloading the results to generate seg hashes if results are directly uploaded to the broadcaster's OS
```

Comment:
If we are always downloading data using `drivers.GetSegmentData()` in the above for loop that means that we should always end up with seg hashes needed for sig verification so we can remove this TODO.

Reply:
done in 2ba85e1c063328389bc03369ed34fb4adc11ae08

Reply:
Not addressing this TODO in this PR sounds good to me. This also gives as an opportunity to see if there is a more efficient way to handle the case where B needs to download the segment data twice from an external OS: once to generate a hash for sig verification and once as a part of the call to the ffmpeg lib that accepts a remote filename and will pull the data. Ideally, B would only download the data once.

Reply:
It seems like https://github.com/livepeer/go-livepeer/commit/2ba85e1c063328389bc03369ed34fb4adc11ae08 removed a different TODO ?

Reply:
oops!  fixed in 8d66157d12d2b4dc0072908d2cf9bacd00266f4a and f558f4e24e1d2a44a0169d55773f5fc037ca2680

Reply:
Actually as-is the TODO:

```
TODO: Consider downloading the results to generate seg hashes if results are directly uploaded to the broadcaster's OS
```

is still relevant because we are not generating seg hashes when B is using an external OS in this PR.

Reply:
reverted - TODO restored in 87d0539812be8b5bae0fffdd5fec6476d3166988